[FLINK-28088] Adjust table store document to catalog (#159)
apache/flink-table-store: docs/content/docs/development/create-table.md

# Create Table

## Catalog

> Table Store uses Catalog to manage all the databases and tables. You need to create a
> Table Store catalog and use it. Table Store catalog has a root directory, configured through `warehouse`.

Reviewer: ```suggestion
Table Store uses its own Catalog to manage all the databases and tables. Users need to configure the type `table-store` and a root directory `warehouse` to use it.
```

```sql
CREATE CATALOG my_catalog WITH (
  'type'='table-store',
  'warehouse'='hdfs://nn:8020/warehouse/path'
);

USE CATALOG my_catalog;
```

Table Store catalog supports SQL DDL commands:
- `CREATE TABLE ... PARTITIONED BY`
- `DROP TABLE ...`
- `ALTER TABLE ...`
- `SHOW DATABASES`
- `SHOW TABLES`

> The path of table in catalog is `${warehouse}/${database_name}.db/${table_name}`.

Reviewer: Suggest moving it right following the sql example

> The path of table in catalog is `${warehouse}/${database_name}.db/${table_name}`.

Reviewer: ```suggestion
This will create a directory under `${warehouse}/${database_name}.db/${table_name}`.
```

Author: This is in the catalog chapter.
